### PR TITLE
Profile backend

### PR DESCRIPTION
**Pull request by lkdfount**
- added unit testing for getUser function that is used to get user info to populate the profile page
- wrote the getUser route in routes
- created getUser in AuthenticationController to find the user matching the email
- removed the "enter site" button from landing page to force users to log in or sign up
- made the profile populated with the user's information
- made the sign up page log the user into the site by saving their email in the store
- created getUser in the AuthenticationService to return the user's information
- added user information to the store

### CODE REVIEW

File: test/unitTests.js [hunk 1]
```diff
@@ -39,35 +38,40 @@ describe('Incorrect email and correct password log in test', function() {
...
-        })
-    });
-
 
 // both tests with correct emails should return a not null user, even if the password is incorrect
 // the password is checked in the AuthenticationService.js file
 describe('Correct email and incorrect password log in test', function() {
- it('Log in should return null with the correct email and incorrect password', async function() {
+ it('Log in should not return null with the correct email and incorrect password', async function() {
        const response = await AuthenticationController.login(correctLogInEmail,incorrectLogInPassword)
```
Comment (rushanniazi) [inline test/unitTests.js:46]: Testing all of these edge cases within our log in unit tests definitely allows us to practice better coding practices and ensures that any additional changes to our code base can be verified through the unit tests before merging to our master branch! Great job

File: travvy-server/app.js [hunk 2]
```diff
@@ -39,3 +39,8 @@ app.post('/login',(req,res) =>{
     message: 'Successful log in'
   })
 })
+app.get('/getUser',(req,res) =>{
+  res.send({
+    message: 'Got user'
```
Comment (rushanniazi) [inline travvy-server/app.js:44]: Nice confirmation message to let the system know that the users profile has been successfully obtained from the users table

File: travvy-server/src/controllers/AuthenticationController.js [hunk 3]
```diff
@@ -48,6 +48,17 @@ module.exports = {
         //retun the user that was found
         //if there is not a user that exists it will return null
         return user
+    },
+    //this method returns a user
```
Comment (rushanniazi) [inline travvy-server/src/controllers/AuthenticationController.js:52]: This function can be later utilized again for any feature that requires the system to grab the profile for a given user! Great

File: travvy/src/components/Profile.vue [hunk 4]
```diff
@@ -1,42 +1,54 @@
...
-    <br>
-    <br>
-    <label>Instagram handle: </label>
-    <input type="text" class="info" v-model="instagram_username"> 
-    <br>
-    <br>
-    <ul>
-      <li><a href="#">Edit profile</a></li>
-    </ul>
+        <h2> Hello {{  getUser().Name }}{{ this.$store.state.name}},</h2>
```
Comment (rushanniazi) [inline travvy/src/components/Profile.vue:4]: Amazing! Storing these fields within the input placeholder will allow us to implement the 'update profile settings' very efficiently

File: travvy/src/store/store.js [hunk 5]
```diff
@@ -26,6 +32,21 @@ export default new Vuex.Store({
...
+        setGender(state,gender){
+            state.gender = gender
+        },
+        setInstagram(state,instagram){
+            state.instagram = instagram
+        },
+        setName(state,name){
+            state.name = name
+        },
+        setActivity(state,activity){
```
Comment (rushanniazi) [inline travvy/src/store/store.js:48]: This can be used later within our search algorithm since it is being kept in the 'store'

**Review by rushanniazi — APPROVED** [review]

File: test/unitTests.js [hunk 6]
```diff
@@ -102,8 +89,22 @@ describe('Recommend Attractions Test', function() {
...
+       assert.notEqual(response, null);
+        })
 
+    });
+//test the getUser function
+describe('Correct email get user test', function() {
+ it('Get user should return null with the an incorrect email', async function() {
+       const response = await AuthenticationController.getUser(incorrectLogInEmail)
+       assert.equal(response, null);
+        })
```
Comment (emilynairnn) [inline test/unitTests.js:105]: You did a great job testing many different cases - very thorough. Well done!

**Review by emilynairnn — APPROVED** [review]

File: travvy/src/components/Profile.vue (hunk 4 above)
Comment (lkdfount) [reply travvy/src/components/Profile.vue:4]: Yes definitely. You will just need to make sure you update things in both the database AND the store for the website to function correctly.

File: travvy/src/components/Profile.vue [hunk 7]
```diff
@@ -1,42 +1,54 @@
...
+        <input type="text" class="info" v-model="preferred_activity_type" :placeholder="[[ this.$store.state.activity ]]"> 
+        <br>
+        <br>
+        <label>Instagram handle: </label>
+        <input type="text" class="info" v-model="instagram_username" :placeholder="[[ this.$store.state.instagram ]]"> 
+        <br>
+        <br>
+        <ul>
+          <li><a href="#">Edit profile</a></li>
+        </ul>
```
Comment (syasrebii) [inline travvy/src/components/Profile.vue:29]: Thanks so much for your thorough updates, Lucy! So exciting to finally be able to see the user information being populated onto the profile page through referencing the store.js file. Also, great use of the "placeholder" attribute to hint suggested values to users, I learned about this attribute thanks to your code!
